theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -45,8 +45,18 @@ class NonlinearOptimizerStatus(Enum):
 @dataclass
 class NonlinearOptimizerInfo(OptimizerInfo):
     converged_iter: torch.Tensor
+    converged_indices: torch.Tensor
+    error_increase_indices: torch.Tensor
```

Comment:
Was the error increase stuff added for the backward support? If not, this might be left over from internal move and was deprecated (for now). cc @luisenp.

Reply:
Yes, we removed this in [this](https://github.com/fairinternal/theseus/commit/1c74f6d578e64993119be5f7214a2eb5c049b89d) commit, due to some issues in the path planning code. The current version does the following:

* When a batch index passes `convergence_check`, the variables values at that index are not updated again.
* For all other indices, variables store the last value after `max_iterations`
* Info returns
  * For all indices (converged or not) the best solution seen (optional)
  * Error history for all iterations (optional)
  * The iteration at which convergence happened (if any)
  * The iteration at which the best solution was reached 

My main question here is whether having `converged_indices` and `error_increase_indices` is necessary for the backward logic introduced in this PR, or if it's some left over from the old parent, as @mhmukadam suggested.

Reply:
I removed `error_increase_indices` and moved `converged_indices` back into the main optimization loop